docs: Update document of cosmovisor (#644)
medibloc/panacea-core: .gitbook/4-validators/4-cosmovisor.md

  * The default value is `none`.
  * `export `DAEMON_RESTART_DELAY=1s`
* DAEMON_DATA_BACKUP_DIR (Optional)
  * Use this option to set a custom backup directory. If not set, $DAEMON_HOME is used.
* UNSAFE_SKIP_BACKUP (Optional)
  * If set to true, upgrades directly without performing a backup. 
  * Otherwise (false, default) backs up the data before trying the upgrade. 
  * The default value of false is useful and recommended in case of failures and when a backup needed to rollback.
  * We recommend using the default backup option UNSAFE_SKIP_BACKUP=false.
* DAEMON_PREUPGRADE_MAX_RETRIES(Optional)
  * The maximum number of times to call pre-upgrade in the application after exit status of 31. 
  * After the maximum number of retries, cosmovisor fails the upgrade.
  * The default value is `0`

We make setting to manually download the binary and to automatically proceed with the upgrade.

## Initialization

Set the necessary environment variables. Below are the default settings for panacea.

```shell
export DAEMON_HOME=$HOME/.panacea
export DAEMON_NAME=panacead
export DAEMON_RESTART_AFTER_UPGRADE=true
export DAEMON_ALLOW_DOWNLOAD_BINARIES=false
```

> Create a directory for the genesis binary. This path should contain the panacead currently running on the mainnet.

Reviewer: ```suggestion
Create a directory for the genesis binary. The current version of binary should be under the directory.
```

cp ./build/panacead $DAEMON_HOME/cosmovisor/genesis/bin
```

Discontinue the currently running `panacead` and run it with `cosmovisor`.

```shell
stop panacead
cosmovisor start
```

## Upgrade chain

cosmovisor supports chain upgrades automatically. This requires some preliminary work.

> You need to create a path in cosmovisor's home path to store the binary to be upgraded.

Reviewer: suggest to use the same word with above
```suggestion
You need to create a directory for the binary to be upgraded.
```

```shell
# Generally, [upgrade-version] refers to the 'Name' value set in the SoftwareUpgradeProposal.
mkdir -p $DAEMON_HOME/cosmovisor/upgrades/[upgrade-version]/bin
```

> Generate a new binary and navigate to the path you created above.

Reviewer: ```suggestion
Generate a new binary and navigate to the directory you created above.
```

```
git clone -b tags/[upgrade-version] https://github.com/medibloc/panacea-core.git
cd panacea-core
make clean && make build
cp ./build/panacead $DAEMON_HOME/cosmovisor/upgrades/[upgrade-version]/bin
```

> The upgrade is complete when the chain is stopped by SoftwareUpgradeProposal, the binaries are replaced by cosmovisor, and the chain is started.s

Reviewer: ```suggestion
The upgrade is complete when the chain reaches the target height of upgrade. The binary is replaced by cosmovisor automatically, and the chain is started.
```
